src/services/nunjucks.js

```diff
@@ -28,51 +30,79 @@ function createEnvironment (context) {
     return '<pre><code>' + string + '</code></pre>';
   });
 
-  addPlugins(env, context);
-
-  return env;
+  addPlugins(env, context, function (err, env) {
+    callback(err, env);
+  });
 }
 
-var addPlugins = function (env, context) {
+var addPlugins = function (env, context, callback) {
   var pluginPath = services.path.getPluginsPath(context);
   try {
     fs.openSync(pluginPath, 'r');
   } catch(e) {
     logger.warn('Unable to find plugins directory at: ' + pluginPath);
-    return;
+    return callback(null, env);
   }
 
   if (!fs.statSync(pluginPath).isDirectory()) {
-    return;
+    return callback(null, env);
   }
 
-  fs.readdirSync(pluginPath).forEach(function (pluginDir) {
-    var plugin;
+  async.eachSeries(fs.readdirSync(pluginPath), function (pluginDir, callback) {
+    var pluginDependencies = [];
 
     try {
-      plugin = require(path.join(pluginPath, pluginDir));
-    } catch(e) {
-      logger.error(e);
-      return;
+      var dependencyDict = JSON.parse(
+        fs.readFileSync(path.join(pluginPath, pluginDir, 'package.json'), 'utf-8')
+      ).dependencies;
+
+      for (var key in dependencyDict) {
+        pluginDependencies.push(key + '@' + dependencyDict[key]);
+      }
+    } catch (e) {
+      pluginDependencies = [];
     }
 
-    if (plugin.templateFilters && plugin.templateFilters.length > 0) {
-      plugin.templateFilters.forEach(function (filter) {
-        var originalFilter = filter[1].bind(env);
-
-        filter[1] = function (input) {
-          try {
-            return originalFilter.apply(env, arguments);
-          } catch(e) {
-            logger.error(e);
-            return input;
-          }
-        };
+    npm.load({}, function () {
```

Comment:
Yeah, it looks like [technically, it can](https://github.com/npm/npm/blob/064d62c2f251cb5e9328f63228f711844b7a1787/lib/config/core.js#L97). This should _probably_ do a `function (err)` check as well, then.